subsys/net/ip/tcp.c

```diff
@@ -960,10 +960,16 @@ static void tcp_send_timer_cancel(struct tcp *conn)
 
 static void tcp_nbr_reachability_hint(struct tcp *conn)
 {
-	if (net_context_get_family(conn->context) == AF_INET6) {
-		net_ipv6_nbr_reachability_hint(
-			net_context_get_iface(conn->context),
-			&conn->dst.sin6.sin6_addr);
+	int64_t now = k_uptime_get();
+	struct net_if *iface = net_context_get_iface(conn->context);
+
+	/* Check that TCP operates on IPv6 and ensure that Neighbor Reachability hints
+	 * are rate-limited (using threshold of half of reachable time).
+	 */
+	if (net_context_get_family(conn->context) == AF_INET6 &&
+	    ((now - conn->last_nd_hint_time) > (net_if_ipv6_get_reachable_time(iface)/2))) {
```

Comment:
We get the uptime and interface for IPv4 packets too which is unnecessary.
I suggest something like
```
int64_t now;
struct net_if *iface;

if (net_context_get_family(conn->context) == AF_INET) {
	return;
}

now = k_uptime_get();
iface = net_context_get_iface(conn->context);

/* ..... */
if ((now - conn->last....
```
and so on

Reply:
Good suggestion, i will apply this.